### PR TITLE
Setup project page

### PR DESCRIPTION
**Pull request by katlavan**

### CODE REVIEW

**Comment by katlavan** [comment]
@loogle18 Please verify

File: app/assets/javascripts/components/chart.es6 [hunk 1]
```diff
@@ -35,11 +35,31 @@ $(()=> {
...
+    fetch(`${url}?repository=${param}`)
+      .then((response) => {
+        return response.json();
+      })
+      .then((labels) => {
+        labels_list.html(labels.html);
+        labels_list.find('select').material_select();
+      })
+      .catch((err) => {
+        console.log('Error : ', err);
```
Comment (ultroned) [inline app/assets/javascripts/components/chart.es6]: Add basic alert or remove this.

File: app/views/setup/index.haml [hunk 2]
```diff
@@ -0,0 +1,9 @@
+.project-selection.js-project-selection{ 'data-url': update_labels_path}
```
Comment (ultroned) [inline app/views/setup/index.haml]: Add space before last bracket.

File: app/views/setup/index.haml [hunk 3]
```diff
@@ -0,0 +1,9 @@
...
+  = select_tag 'backlog', options_for_select(@repositories), id: :project_selection
```
Comment (ultroned) [inline app/views/setup/index.haml:2]: Why not class instead id?

File: app/controllers/setup_controller.rb [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+  def update_labels
+    @labels = LabelsCarrier.new(load_events)
+
+    content = load_labels_content(@labels)
+    render json: { html: content}
+  end
+
+  private
+
+  def load_labels_content(labels)
```
Comment (ultroned) [inline app/controllers/setup_controller.rb]:
```
if load_events.exists?
    render_to_string( partial: 'setup/labels' ,  collection: @labels.label_status, as: :label,
                      locals: { select_options: @labels.labels_name_list })
else
    I18n.t('no_labels')
end
```

File: app/controllers/setup_controller.rb [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+    if labels.uniq_labels.nil?
+      I18n.t('no_labels')
+    else
+      render_to_string( partial: 'setup/labels' ,  collection: labels.label_status, as: :label,
+                        locals: { select_options: labels.labels_name_list })
+    end
+  end
+
+  def load_repositories
```
Comment (ultroned) [inline app/controllers/setup_controller.rb]: It's duplication, please move to protected method of application controller, and just call it.
And remove this logic from metrics controller as well.

File: app/controllers/setup_controller.rb [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+    Github.new(oauth_token: session[:token]).repos.list.map(&:name)
+  end
+
+  def load_events
```
Comment (ultroned) [inline app/controllers/setup_controller.rb]: It's duplication, please move to protected method of application controller, and just call it.
And remove this logic from metrics controller as well.

File: app/controllers/setup_controller.rb [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+    if params[:repository].present?
+      Event.where(repo: params[:repository])
```
Comment (ultroned) [inline app/controllers/setup_controller.rb]: And add `created` params like [here](https://github.com/jetthoughts/gh_metrics/pull/4/files#diff-12435b1e4e29226967cae794fbffb44cR21)

**Review by ultroned — COMMENTED** [review]
Please do rebase if you did not it and consider my comments.
@m-k please check it too. I could mess something.

**Comment by katlavan** [comment]
@loogle18 Fixed. Please recheck

File: app/views/layouts/_topbar.haml [hunk 8]
```diff
@@ -21,12 +13,8 @@
           %a.dropdown-button{'chartLineData-activates': 'dropdown1', href: "#!"}
             Repositories
             %i.material-icons.right
-      - if login?
-        %li
-          = link_to 'Logout', logout_path, method: :destroy, class: 'right'
+        %li= link_to 'Setup Repositrory', setup_path, class: 'right'
```
Comment (m-k) [inline app/views/layouts/_topbar.haml]: Setup Reposit~r~ory 🇬🇧 

Please move all texts to translations.

Also, this button should move us to editing current repository or be `Setup repositories`.

File: app/views/setup/index.haml [hunk 9]
```diff
@@ -0,0 +1,9 @@
+.project-selection.js-project-selection{ 'data-url': update_labels_path }
+  = select_tag 'backlog', options_for_select(@repositories), id: :project_selection
+.row.js-labels-list
+  - if @labels.uniq_labels.nil?
+    .col.s12= t('no_labels')
+  - else
+    = render partial: 'setup/labels', collection: @labels.label_status, as: :label,
+                                locals: { select_options: @labels.labels_name_list }
```
Comment (m-k) [inline app/views/setup/index.haml]: Please fix formatting.

**Review by m-k — CHANGES_REQUESTED** [review]
@katlavan please consider by comments.

**Comment by katlavan** [comment]
@m-k Fixed. Please recheck.

File: app/controllers/setup_controller.rb [hunk 10]
```diff
@@ -0,0 +1,25 @@
+class SetupController < ApplicationController
+
+  def index
+    @repositories = load_repositories
+    @labels = LabelsCarrier.new(load_events)
+
```
Comment (m-k) [inline app/controllers/setup_controller.rb]: please remove empty line

File: app/carriers/labels_carrier.rb [hunk 11]
```diff
@@ -1,6 +1,11 @@
 class LabelsCarrier
+  attr_reader :label_status
+
+  LABEL_STATUS = %w(backlog ready progress verify codereview)
```
Comment (m-k) [inline app/carriers/labels_carrier.rb]: Freeze mutable objects assigned to constants

File: app/carriers/labels_carrier.rb [hunk 12]
```diff
@@ -10,4 +15,8 @@ def uniq_labels
       end.uniq
     end
   end
+
+  def labels_name_list
+    uniq_labels.map{|l| l[:name]}
```
Comment (m-k) [inline app/carriers/labels_carrier.rb]: Please add spaces inside `{}`

**Comment by katlavan** [comment]
@m-k Done. Please recheck

File: app/carriers/labels_carrier.rb [hunk 13]
```diff
@@ -1,6 +1,11 @@
 class LabelsCarrier
+  attr_reader :label_status
+
+  LABEL_STATUS = %w(backlog ready progress verify codereview)
+
   def initialize(events)
     @events = events
+    @label_status = LABEL_STATUS
```
Comment (m-k) [inline app/carriers/labels_carrier.rb]: No need to use instance variable, please use `LabelsCarrier::LABEL_STATUS` in views.

File: app/controllers/setup_controller.rb [hunk 14]
```diff
@@ -0,0 +1,24 @@
+class SetupController < ApplicationController
+
```
Comment (m-k) [inline app/controllers/setup_controller.rb]: ✂️

File: app/controllers/setup_controller.rb [hunk 15]
```diff
@@ -0,0 +1,24 @@
...
+    render json: { html: load_labels_content }
+  end
+
+  private
+
+  def load_labels_content
+    if load_events.exists?
+      render_to_string( partial: 'setup/labels' ,  collection: @labels.label_status, as: :label,
+                        locals: { select_options: @labels.labels_name_list })
```
Comment (m-k) [inline app/controllers/setup_controller.rb]: please remove spaces
```
render_to_string( partial: 'setup/labels' ,  collection: @labels.label_status, as: :label,
                 ^                       ^ ^
```

File: config/locales/en.yml [hunk 16]
```diff
@@ -20,9 +20,15 @@
...
   errors:
     messages:
       network_error: 'Network error: please try again later'
+  no_labels: 'No events for current repository'
+  buttons:
+    setup_repositories: "Setup Repositories"
+    logout: "Logout"
+  labels:
+    repositories: "Repositories"
+    app_name: "Github Metrics"
```
Comment (m-k) [inline config/locales/en.yml]: Prefer single-quoted strings when you don't need string interpolation or special symbols.

**Review by m-k — CHANGES_REQUESTED** [review]
@katlavan please consider my comments.

**Comment by katlavan** [comment]
@m-k Done. Please recheck.

**Review by m-k — APPROVED** [review]

**Pull request merged**